### PR TITLE
CAMEL-12541: rsClient does not work programmatically, only with XML

### PR DESCRIPTION
**Pull request by willianantunes**
When the endpoint is created with the bean (see [here](https://github.com/apache/camel/blob/39c0d63d923bfe9236834ecb1c4470bb7e9e7eaa/components/camel-cxf/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsComponent.java#L76) how it's found) it is always available and is used for the _consumer approach_ (see the method [newJAXRSServerFactoryBean](https://github.com/apache/camel/blob/39c0d63d923bfe9236834ecb1c4470bb7e9e7eaa/components/camel-cxf/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsSpringEndpoint.java#L65)). 

Now the [same behavior](https://github.com/willianantunes/camel/blob/30a4bdbf0975c8efc06d504b751e88ec4af5e64a/components/camel-cxf/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsSpringEndpoint.java#L71) is used for _producer approach_ as nowadays [we always return a new instance of SpringJAXRSClientFactoryBean](https://github.com/apache/camel/blob/39c0d63d923bfe9236834ecb1c4470bb7e9e7eaa/components/camel-cxf/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsSpringEndpoint.java#L71).

### CODE REVIEW

**Comment by willianantunes** [comment]
Before creating this PR I'd run the tests and re-ran after inserting the fix to make sure if it was OK. 

I believe [CxfRsProducerAddressOverrideTest.testAddressMultiOverride](https://github.com/apache/camel/blob/39c0d63d923bfe9236834ecb1c4470bb7e9e7eaa/components/camel-cxf/src/test/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducerAddressOverrideTest.java#L155) is now breaking because LRUSoftCache is return the wrong stored object [here](https://github.com/apache/camel/blob/39c0d63d923bfe9236834ecb1c4470bb7e9e7eaa/components/camel-cxf/src/main/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducer.java#L862) when the test execute [line 166](https://github.com/apache/camel/blob/39c0d63d923bfe9236834ecb1c4470bb7e9e7eaa/components/camel-cxf/src/test/java/org/apache/camel/component/cxf/jaxrs/CxfRsProducerAddressOverrideTest.java#L166).

**Comment by willianantunes** [comment]
When the cache works properly it has the following:
```
{
    http://localhost:1101/CxfRsProducerNonExistingAddressOverrideTest=org.apache.camel.component.cxf.spring.SpringJAXRSClientFactoryBean@625487a6, 
    http://localhost:1101/CxfRsProducerAddressOverrideTest=org.apache.camel.component.cxf.spring.SpringJAXRSClientFactoryBean@36224f93
}
```
See that each value is a different object (625487a6 and 36224f93). But when I run a test with this PR I get the following:
```
{
    http://localhost:1101/CxfRsProducerNonExistingAddressOverrideTest=org.apache.camel.component.cxf.spring.SpringJAXRSClientFactoryBean@ab58eea1, 
    http://localhost:1101/CxfRsProducerAddressOverrideTest=org.apache.camel.component.cxf.spring.SpringJAXRSClientFactoryBean@ab58eea1
}
```
Both keys have the same object. That's why the test is breaking.

File: components/camel-cxf/src/test/java/org/apache/camel/component/cxf/jaxrs/CxfRsSpringEndpointTest.java [hunk 1]
```diff
@@ -62,13 +69,51 @@ public void testCreateCxfRsClientFactoryBean() {
...
+        return clientFactoryBean;
+    }    
+    
     @Override
     protected AbstractXmlApplicationContext createApplicationContext() {
         String version = Version.getCurrentVersion();
+        
+        ClassPathXmlApplicationContext applicationContext = null;
+        
         if (version.startsWith("2") && (version.contains("2.5") || version.contains("2.4"))) {
```
Comment (WillemJiang) [inline components/camel-cxf/src/test/java/org/apache/camel/component/cxf/jaxrs/CxfRsSpringEndpointTest.java]: This version check could be removed as we already move to 2.6.x

File: components/camel-cxf/src/test/java/org/apache/camel/component/cxf/jaxrs/CxfRsSpringEndpointTest.java [hunk 2]
```diff
@@ -62,13 +69,51 @@ public void testCreateCxfRsClientFactoryBean() {
 
     }
     
+    @Test
+    public void testCreateCxfRsClientFactoryBeanProgrammatically() {
+        
+        CxfRsEndpoint endpoint = resolveMandatoryEndpoint("cxfrs://bean://" + BEAN_SERVICE_ENDPOINT_NAME, CxfRsEndpoint.class);
+        SpringJAXRSClientFactoryBean cfb = (SpringJAXRSClientFactoryBean)endpoint.createJAXRSClientFactoryBean();
```
Comment (WillemJiang) [inline components/camel-cxf/src/test/java/org/apache/camel/component/cxf/jaxrs/CxfRsSpringEndpointTest.java]: we may need to check the instance of cfb is not the one we set before.

**Pull request merged**